subsys/bluetooth/controller/ll_sw/ull_llcp_conn_upd.c

```diff
@@ -787,10 +860,31 @@ static void rp_cu_send_conn_update_ind(struct ll_conn *conn, struct proc_ctx *ct
 	if (llcp_rr_ispaused(conn) || !llcp_tx_alloc_peek(conn, ctx)) {
 		ctx->state = RP_CU_STATE_WAIT_TX_CONN_UPDATE_IND;
 	} else {
-		cu_prepare_update_ind(conn, ctx);
-		rp_cu_tx(conn, ctx, PDU_DATA_LLCTRL_TYPE_CONN_UPDATE_IND);
-		ctx->rx_opcode = PDU_DATA_LLCTRL_TYPE_UNUSED;
-		ctx->state = RP_CU_STATE_WAIT_INSTANT;
+		/* ensure alloc of TX node, before possibly waiting for NTF node */
+		ctx->node_ref.tx_ack = llcp_tx_alloc(conn, ctx);
+		if (ctx->node_ref.rx == NULL && !llcp_ntf_alloc_is_available()) {
+			/* No RX node piggy, and no NTF avail, so go wait for one, before TX'ing */
+			ctx->state = RP_CU_STATE_WAIT_NTF_AVAIL;
+		} else {
+			rp_cu_send_conn_update_ind_finalize(conn, ctx, evt, param);
+		}
+	}
+}
+
+static void rp_cu_st_wait_ntf_avail(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt,
+				    void *param)
+{
+	switch (evt) {
+	case LP_CU_EVT_RUN:
+		if (llcp_ntf_alloc_is_available()) {
+			/* If NTF node is now avail, so pick it up and continue */
+			ctx->node_ref.rx = llcp_ntf_alloc();
```

Comment:
No need to allocate node here, as it's done in `rp_cu_send_conn_update_ind_finalize()`

Reply:
Right - also no reason to use LP_CU_EVT_RUN define instead of RP_CU_EVT_RUN!!